src/ansi-c/library/stdlib.c

```diff
@@ -115,23 +115,43 @@ inline void *malloc(__CPROVER_size_t malloc_size)
   // realistically, malloc may return NULL,
   // and __CPROVER_allocate doesn't, but no one cares
   __CPROVER_HIDE:;
-  void *malloc_res;
-  malloc_res = __CPROVER_allocate(malloc_size, 0);
 
-  // make sure it's not recorded as deallocated
-  __CPROVER_deallocated=(malloc_res==__CPROVER_deallocated)?0:__CPROVER_deallocated;
+    if(__CPROVER_malloc_failure_mode == 1)
```

Comment:
In an ideal world :-) rather than magic numbers here for the failure modes, these would be static named constants. Bonus points of those named constants were exposed to user programs as __CPROVER_blah constants.

Reply:
Done.